src/main/java/org/radarbase/appserver/service/FcmNotificationService.java

```diff
@@ -1,22 +1,311 @@
+/*
+ *
+ *  *
+ *  *  * Copyright 2018 King's College London
+ *  *  *
+ *  *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  *  * you may not use this file except in compliance with the License.
+ *  *  * You may obtain a copy of the License at
+ *  *  *
+ *  *  *   http://www.apache.org/licenses/LICENSE-2.0
+ *  *  *
+ *  *  * Unless required by applicable law or agreed to in writing, software
+ *  *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  *  * See the License for the specific language governing permissions and
+ *  *  * limitations under the License.
+ *  *  *
+ *  *
+ *
+ */
+
 package org.radarbase.appserver.service;
 
-import org.radarbase.appserver.dto.fcm.FcmNotifications;
+import lombok.SneakyThrows;
+import org.radarbase.appserver.converter.ConverterFactory;
+import org.radarbase.appserver.entity.Notification;
+import org.radarbase.appserver.entity.Project;
+import org.radarbase.appserver.entity.User;
+import org.radarbase.appserver.entity.UserMetrics;
+import org.radarbase.appserver.exception.AlreadyExistsException;
+import org.radarbase.appserver.exception.InvalidNotificationDetailsException;
+import org.radarbase.appserver.exception.InvalidUserDetailsException;
+import org.radarbase.appserver.exception.NotFoundException;
+import org.radarbase.appserver.repository.NotificationRepository;
+import org.radarbase.appserver.repository.ProjectRepository;
+import org.radarbase.appserver.repository.UserRepository;
+import org.radarbase.appserver.service.scheduler.NotificationSchedulerService;
+import org.radarbase.fcm.dto.FcmNotificationDto;
+import org.radarbase.fcm.dto.FcmNotifications;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicReference;
 
+/**
+ * {@link Service} for interacting with the {@link Notification} {@link javax.persistence.Entity} using the
+ * {@link NotificationRepository}.
+ *
+ * @author yatharthranjan
+ */
+@Service
 public class FcmNotificationService {
 
+    @Autowired
+    private NotificationRepository notificationRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ProjectRepository projectRepository;
+
+    @Autowired
+    private NotificationSchedulerService schedulerService;
+
+    @Transactional(readOnly = true)
     public FcmNotifications getAllNotifications() {
-        return null;
+        List<Notification> notifications = notificationRepository.findAll();
+        return new FcmNotifications().setNotifications(ConverterFactory.getNotificationConverter().entitiesToDtos(notifications));
```

Comment:
We use Google Code Styles for most of the projects. Maybe something to consider

Reply:
Yes was planning to add checkstyle and pmd in a later PR